lms/djangoapps/course_goals/views.py

```diff
@@ -106,3 +108,12 @@ def emit_course_goal_event(sender, instance, **kwargs):
             'goal_key': instance.goal_key,
         }
     )
+    update_google_analytics(name, instance.user.id)
```

Comment:
I think you probably need to add this before `tracker.emit`. I have no issue if it's working fine here as well.